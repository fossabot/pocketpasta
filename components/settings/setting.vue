<template>
  <li class="list-group-item">
    <div class="row">
      <div class="col-md-6">
        <v-list-tile>
          <v-list-tile-action>
            <v-icon>{{ icon }}</v-icon>
          </v-list-tile-action>
          <v-list-tile-content>
            <v-list-tile-title>
              <label>{{ label }}</label>
            </v-list-tile-title>
            <span class="form-text text-muted">
              {{ description }}
            </span>
          </v-list-tile-content>
        </v-list-tile>
      </div>
      <div class="col-md-6 align-self-center">
        <slot>
          <input type="text" class="form-control" placeholder="placeholder" />
        </slot>
      </div>
    </div>
  </li>
</template>

<script>
export default {
  props: {
    icon: { type: String, default: null },
    label: { type: String, required: true },
    description: { type: String, default: null },
  },
};
</script>
