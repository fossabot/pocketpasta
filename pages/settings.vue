<template>
  <div class="container">
    <h1>{{ $t('settings.heading') }}</h1>
    <div class="card">
      <div class="card-header text-uppercase">Style</div>
      <ul class="list-group list-group-flush">
        <setting
          icon="style"
          label="Theme"
          description="Choose a bootstrap CSS theme to style the app your way."
        >
          <theme-selector />
        </setting>
        <setting
          icon="language"
          label="Language"
          description="What language do you speak?"
        >
          <LangSwitcher />
        </setting>
      </ul>
    </div>
  </div>
</template>

<script>
import ThemeSelector from '@/components/ThemeSelector.vue';
import LangSwitcher from '@/components/Layout/LangSwitcher';
import setting from '@/components/settings/setting';

export default {
  components: {
    ThemeSelector,
    LangSwitcher,
    setting,
  },
  head() {
    return {
      link: [
        {
          hid: 'canonical',
          rel: 'canonical',
          href: 'https://pocketpasta.com/settings',
        },
      ],
    };
  },
};
</script>
