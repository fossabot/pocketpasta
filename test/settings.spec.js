import { shallowMount } from '@vue/test-utils';
import settings from '@/pages/settings.vue';

const factory = () => shallowMount(settings, {});

describe('settings', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
