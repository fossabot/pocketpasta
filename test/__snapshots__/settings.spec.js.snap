// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings renders properly 1`] = `"<div class=\\"container\\"><h1>settings.heading</h1> <div class=\\"card mb-3\\"><h2 class=\\"subheading card-header text-uppercase\\">Style</h2> <ul class=\\"list-group list-group-flush\\"><setting-stub icon=\\"style\\" label=\\"Theme\\" description=\\"Choose a bootstrap CSS theme to style the app your way.\\" id=\\"theme-label\\"><theme-selector-stub aria-labelledby=\\"theme-label\\"></theme-selector-stub></setting-stub></ul></div> <div class=\\"card mb-3\\"><h2 class=\\"subheading card-header text-uppercase\\">Content</h2> <ul class=\\"list-group list-group-flush\\"><setting-stub icon=\\"language\\" label=\\"Language\\" description=\\"What language do you speak?\\"><langswitcher-stub></langswitcher-stub></setting-stub></ul></div></div>"`;
